azure/storage-account/main.tf

```diff
@@ -18,13 +18,28 @@ resource "azurerm_storage_account" "main" {
   tags                          = var.tags
 
   # This is a recommendation from BridgeCrew to enable queue logging for storage accounts.
-  queue_properties {
-    logging {
-      delete                = true
-      read                  = true
-      write                 = true
-      version               = "1.0"
-      retention_policy_days = 10
+  dynamic "queue_properties" {
+    for_each = var.enable_queue ? [1] : []
+    content {
+      logging {
+        delete                = var.queue_properties.logging.delete
+        read                  = var.queue_properties.logging.read
+        write                 = var.queue_properties.logging.write
+        version               = var.queue_properties.logging.version
+        retention_policy_days = var.queue_properties.logging.retention_policy_days
+      }
+    }
+  }
+
+  dynamic "blob_properties" {
+    for_each = var.enable_blob ? [1] : []
```

Comment:
Same as above.

Reply:
I have a similar code block in aurora and had to use a string for I assume the same dynamic-ish single input concept...

https://github.com/massdriver-cloud/aws-aurora-postgresql/blob/56806b2df11dc939a51fcc19eec5dc2ce0fd9b4a/src/cluster.tf#L39